File: pi_approximation_by_throwing_darts.ex
defmodule PiApproximationByThrowingDarts do
  def pi_approx(sampling) do
    :random.seed(:erlang.now)
    dart = fn(n) ->
      [x, y] = [:random.uniform, :random.uniform]
      if :math.pow(x, 2) + :math.pow(y, 2) < 1 do 1 else 0 end
    end
    count = 1..sampling |> Enum.map(dart) |> Enum.sum
    count / sampling * 4
  end
end

sampling = if length(System.argv) == 1 do
  System.argv |> List.first |> String.to_integer
else
  1000000
end
IO.puts "π ≒ #{PiApproximationByThrowingDarts.pi_approx(sampling)}"

File: PiApproximationByThrowingDarts.swift
class PiApproximationByThrowingDarts {
  static func piApprox(sampling: Int) -> Double {
    let dart = { (n: Int) -> Int in
      let (x, y) = (drand48(), drand48())
      return pow(x, 2) + pow(y, 2) < 1 ? 1 : 0
    }
    let count = (1...sampling).map(dart).reduce(0, combine: +)
    return Double(count) / Double(sampling) * 4
  }
}

let sampling = 1000000
print("π ≒ \(PiApproximationByThrowingDarts.piApprox(sampling))")
